src/utilities/utilities.hpp

```diff
@@ -28,6 +28,21 @@
 #include "utilities/clblast_exceptions.hpp"
 #include "utilities/msvc.hpp"
 
+#define FULL_SEARCH_STRATEGY 0
+#define RANDOM_SEARCH_STRATEGY 1
+#define PSO_STRATEGY 2
+#define DVDT_STRATEGY 3
+
+#define DEFAULT_STRATEGY 0
+
+#define DEFAULT_PSO_SWARM 8 
+#define DEFAULT_PSO_G 0.3
+#define DEFAULT_PSO_L 0.6
+#define DEFAULT_PSO_R 0.1
```

Comment:
Could you try to avoid using pre-processor defines? With C++ and especially since C++11 it shouldn't be necessary to use these unfriendly things anymore ;-) You should probably move them to wherever they belong (e.g. in the tuner) and declare them as `constexpr` instead.